pyvrp/search/LocalSearch.py

```diff
@@ -76,7 +76,7 @@ def get_neighbours(self) -> List[List[int]]:
         """
         return self._ls.get_neighbours()
 
-    def run(
+    def __call__(
         self,
         solution: Solution,
         cost_evaluator: CostEvaluator,
```

Comment:
I want to see if we can move this down into C++, because that'd avoid loading/unloading solutions more than needed, and allow this to be done more efficiently (no need to get the cost evaluator involved, for example).

Only difficulty is shuffling, but I'll tinker with that a bit to figure something out.

Reply:
Shuffling once before starting an (otherwise deterministic) loop of search/intensify seems to be enough randomisation (for myself: job ID 3230968):
```
 Avg. objective: 333438.6 (min: 333248, max: 333589)
Avg. iterations: 55452.5 (min: 45535, max: 62129)
```
This is extremely similar to current main, so this direction could work.

Reply:
I've implemented this in a branch on my own fork. I'm running a job overnight (ID: 3232120) to see if that's similar to what I found above. If so I'll open a PR tomorrow.

Reply:
Ooh, something seems to be broken if we don't import/export every time. Cool! I'll see if I can debug that later this week.

Reply:
Why do we need to move this to c++? I agree loading/unloading is unnecessary but 1) is that really a bottleneck and 2) we can also fix that by binding the loading/unloading functions seperately, so you can just do that once.

Binding load/export seperately would make the c++ api less robust (i.e. you need to load the solution before searching just like you need to set the neighbourhood), but I consider that internal as we expose the Python API and this way we keep all the flexibility at the Python level making it much easier for users to adapt (e.g. doing a max number of iterations, only intensify sometimes, etc.).

Reply:
We already have all that in #298 without having to break encapsulation.

Reply:
I don't understand? In #298 the search loop is in C++, so if a user wants to customize it from Python he/she first needs to implement it in Python again and then will incur the overhead of loading/exporting solutions repeatedly which you aim to resolve. What do you mean with 'we already have all that' then?

I think moving things back from Python to C++ for no clear reason is not the direction we should go. Your primary reason seems efficiency which is unclear if it's a bottleneck?

Reply:
I'm going to copy some of this to #298 because it's off topic here.